addon/components/x-form.js

```diff
@@ -71,18 +71,20 @@ export default Ember.Component.extend({
 
   init() {
     this._super(...arguments);
-    this._setUpChangeset();
+    if(!this.get('data')) {
+      throw new Error('x-form needs data');
+    }
```

Comment:
Is this necessarily true? In previous incarnations, what we did was to make the this the differentatior between "new" forms and "edit" forms. A "new" form _has_ no `data`, whereas an "edit" for represents a change to an existing identity and so will have _data_